### PR TITLE
Emad/controlling member crc repositroy

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

- Added repositroy and logic for Crc controlling member

### CODE REVIEW

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 1]
```diff
@@ -13,19 +14,30 @@ internal class ControllingMemberCrcAppManager :
...
+    private readonly IBizLicApplicationRepository _bizLicApplicationRepository;
+    private readonly IControllingMemberCrcRepository _controllingMemberCrcRepository;
 
-    public ControllingMemberCrcAppManager(IMapper mapper)
+
+    public ControllingMemberCrcAppManager(IMapper mapper,
+        IBizLicApplicationRepository bizApplicationRepository,
+        IControllingMemberCrcRepository controllingMemberCrcRepository)
     {
+        _bizLicApplicationRepository = bizApplicationRepository;
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:25]: why we need this?

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                .Expand(a => a.spd_ApplicantId_contact)
+                .Expand(a => a.spd_ApplicantId_account)
+                .Where(a => a.spd_applicationid == cmd.ParentBizLicApplicationId)
+                .SingleOrDefaultAsync(ct);
+
+        if (bizLicApplication == null)
+            throw new ArgumentException("Original business licence application was not found.");
+
+        // create controlling member application
+        spd_application app = _mapper.Map<spd_application>(cmd);
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:42]: where is mapping?

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        _context.AddTospd_applications(app);
+        // create contact
+        contact contact = _mapper.Map<contact>(cmd);
+        contact.contactid = Guid.NewGuid();
+        _context.AddTocontacts(contact);
+
+        await _context.SaveChangesAsync(ct);
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:49]: do not do saveChanges here

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            .Where(a => a.accountid == bizLicApplication.spd_ApplicantId_account.accountid)
+            .Where(a => a.statecode == DynamicsConstants.StateCode_Active)
+            .FirstOrDefault();
+        if (account != null)
+        {
+            _context.SetLink(app, nameof(spd_application.spd_OrganizationId), account);
+        }
+
+        _context.SetLink(app, nameof(app.spd_ApplicantId_contact), contact);
+        await _context.SaveChangesAsync(ct);
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:61]: do not do saveChanges here

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+        //create the aliases
+        foreach (var item in cmd.Aliases)
+        {
+            AddAlias(item, contact);
+        }
+
+        _context.AddLink(bizLicApplication, nameof(bizLicApplication.spd_businessapplication_spd_workerapplication), app);
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:69]: I am not sure this part. Have you talked this with Kenny? I guess it should be in bizContact entity. Not sure though

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 2 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:42]: you mean the file? in the ControllingMemberCrcApplication directory, under Repository.

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 5 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:69]: yes, he confirmed this links the child to the parent

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 4 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:61]: removed (in dynamics, before we want to add or set a link, it does not matter that the object has saved and has a primary value?)

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 3 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:49]: removed

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 2 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:42]: yes

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs (hunk 4 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs]: yeah, as long as you already addToxxx, Please test if you remove saveChanges, it is still working normally

**Review by peggy-quartech — APPROVED** [review]

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**